frappe/desk/doctype/number_card/number_card.json

```diff
@@ -39,7 +39,7 @@
  "fields": [
   {
    "depends_on": "eval: ['Document Type', 'Custom'].includes(doc.type)",
-   "description": "Restrict visibility to users with read access to this doctype.",
+   "description": "This card is visible only to Administrator and System Managers by default. Set a DocType to share with users who have read access.",
```

Comment:
this description is relevant only if the card type is "Custom". For the card type "Document Type", the field is already mandatory and impacts number query in addition to the use-case described here.